### ISSUE TITLE
Invalid version specification for tools does not raise an error

### ISSUE THREAD

**Issue by driskell** [shivammathur/setup-php#734]
**Describe the bug**
When specifying `composer:'2.4'` it will still install `2.5.7` and not complain (the quotes make the version invalid I think)

**Version**
- [x] I have checked releases, and the bug exists in the latest patch version of `v1` or `v2`.
- [x] `v2`
- [ ] `v1`

**Runners**

- [x] GitHub Hosted
- [ ] Self Hosted

**Operating systems**

**PHP versions**

8.1

**To Reproduce**
```
- name: Setup PHP with composer v2
  uses: shivammathur/setup-php@v2
  with:
    php-version: '8.1'
    tools: composer:'2.4'
```

**Expected behavior**

Report invalid version (or install 2.4)

**Screenshots/Logs**

```
Run shivammathur/setup-php@v2
  with:
    php-version: 8.1
    tools: composer:'2.4'
    ini-file: production
/usr/bin/bash /home/runner/work/_actions/shivammathur/setup-php/v2/src/scripts/run.sh

==> Setup PHP
✓ PHP Updated to PHP 8.1.18

==> Setup Tools
✓ composer Added composer 2.5.7

==> #StandWithUkraine
✓ read-more https://setup-php.com/support-ukraine
```

**Additional context**

**Are you willing to submit a PR?**

**Comment by shivammathur** [comment]
Fixed in https://github.com/shivammathur/setup-php/commit/5a3381c35bc7c821e40293b14972c2b590a5fb3e. Will be in the next release

**Comment by shivammathur** [comment]
Released 2.25.3 with the fix.
https://github.com/shivammathur/setup-php/releases/tag/2.25.3
